### PR TITLE
`Assessment`: Make example submissions available for editors

### PR DESCRIPTION
**Pull request by mallgaier**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
When Editors try to create an example submission used for assessment training, a 404 is thrown and the editor doesn't have access to the assessed example submission.

### Description
On the server side, there is currently only a distinction between Instructor and Tutor. While the Instructor always gets the Example Assessment, a 404 is thrown for the Tutor and Assessment Training. The Tutor only gets the Example Assessment, if it is not used for Assessment Training.
This PR only makes a distinction between editor and tutor. Editors (and subsequently Instructors) always get the example submission, while tutors only get the example submission, if it is not used for assessment training.

Also updated the documentation: https://artemis-platform--5566.org.readthedocs.build/en/5566/admin/accessRights/

### Steps for Testing

Prerequisites:
- 1 Editor

1. Log in to Artemis
2. Navigate to Course Administration -> Exercise
3. Create a new Modelling Exercise
4. Create a Example Submission used for assessment training
5. Store the example submission, navigate one page back and again into the example submission
6. Verify, that no Error (404) is thrown and that the example assessment is visible

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by Dominik-Weinzierl — APPROVED** [review]
Tested on ts1, code changes look als good :+1:

**Review by cathrinrl — DISMISSED** [review]
Tested with User 15 on TS1. Changes looked good (y)

File: src/main/java/de/tum/in/www1/artemis/web/rest/AssessmentResource.java [hunk 1]
```diff
@@ -171,13 +171,13 @@ protected ResponseEntity<Result> getExampleAssessment(long exerciseId, long subm
         final var exampleSubmission = exampleSubmissionRepository.findBySubmissionIdWithResultsElseThrow(submissionId);
 
         var user = userRepository.getUserWithGroupsAndAuthorities();
-        var isAtLeastInstructor = authCheckService.isAtLeastInstructorForExercise(exercise, user);
+        var isAtLeastEditor = authCheckService.isAtLeastEditorForExercise(exercise, user);
         var isAtLeastTutor = authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user);
         // It is allowed to get the example assessment, if the user is an instructor or
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/web/rest/AssessmentResource.java:176]:
```suggestion
        // It is allowed to get the example assessment, if the user is at least an editor or
```

**Review by Dominik-Weinzierl — APPROVED** [review]

**Review by Dominik-Weinzierl — APPROVED** [review]
Looks good now, thanks!

**Review by Dominik-Weinzierl — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**